src/image-service.ts

```diff
@@ -43,18 +49,49 @@ export const ImageService = {
     }
 
     const auth = {} // Images are public so not authentication info is required
-    await this.fetchImage(imageName, auth, docker)
+    await this.fetchImageWithRetry(imageName, auth, docker)
   },
 
-  /* Retrieve the imageName using the auth details provided, if any */
-  async fetchImage(
+  /* Retrieve the image using the auth details provided, if any with retry and backoff */
+  async fetchImageWithRetry(
     imageName: string,
     auth = {},
     docker = new Docker()
   ): Promise<void> {
-    core.info(`Pulling image ${imageName}...`)
-    const stream = await docker.pull(imageName, {authconfig: auth})
-    await endOfStream(docker, new Readable().wrap(stream))
-    core.info(`Pulled image ${imageName}`)
+    let attempt = 0
+
+    while (attempt < MAX_RETRIES) {
+      try {
+        core.info(`Pulling image ${imageName} (attempt ${attempt + 1})...`)
+        const stream = await docker.pull(imageName, {authconfig: auth})
+        await endOfStream(docker, new Readable().wrap(stream))
+        core.info(`Pulled image ${imageName}`)
+        return // Exit on success
+      } catch (error) {
+        attempt++
+
+        if (
+          error instanceof Error &&
+          (error.message.includes('429') ||
+            error.message.toLowerCase().includes('too many requests'))
+        ) {
+          const delay = INITIAL_DELAY_MS * Math.pow(2, attempt) // Exponential backoff
```

Comment:
can we add a jitter too? That way the backoffs all don't start at the same time

Reply:
Also we should consider adding tests too?

Reply:
I am in the process of adding test the mocking of Docker API is making it difficult and leading to failure.

Reply:
Thank you for the review. Jitter is added!

Reply:
> Also we should consider adding tests too?

Tests are added too!